sound/soc/intel/common/soc-acpi-intel-cnl-match.c

```diff
@@ -60,6 +162,20 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_cnl_sdw_machines[] = {
 		.sof_fw_filename = "sof-cnl.ri",
 		.sof_tplg_filename = "sof-cnl-rt5682-sdw2.tplg"
 	},
+	{
+		.link_mask = 0xF,
+		.links = up_extreme_2amps_dummy,
+		.drv_name = "sof_sdw",
+		.sof_fw_filename = "sof-cnl.ri",
+		.sof_tplg_filename = "sof-cml-rt711-rt1308-rt715.tplg",
+	},
+	{
+		.link_mask = 0xF,
```

Comment:
only 3 links enabled,  0xB ?

Reply:
but why only 3 amps for this config?

Reply:
one headset codec, one amp, one mic codec.

Reply:
@plbossart the mono tplg gets picked only if the ordering between the up_extreme_1amp_dummy is first in this array. Otherwise, I see that sof-cml-rt711-rt1308-rt715.tplg is the one that gets loaded. What am I doing wrong?

Reply:
I changed the bit mask. 

@ranj063 the ordering is needed, we first want to check if we find the configuration with two amps, and fallback to the 1amp if  the second amp is not found. If we revert the order, we will always use the single amp config.